### PR TITLE
Add/Update TF controls

### PR DESCRIPTION
**Pull request by rajlearner17**
### Checklist
- [ ] Issue(s) linked

1. [x] [Updated eks_cluster_endpoint_restrict_public_access](https://github.com/turbot/steampipe-mod-terraform-aws-compliance/issues/78)
2. [x] cloudfront_distribution_origin_access_identity_enabled query fails due SQL error #77
3. [x] ec2_ami_launch_permission_restricted
4. [x] apigateway_method_restricts_open_access
5. [x] apigateway_domain_name_use_latest_tls
6. [x] cloudfront_response_header_use_strict_transport_policy_setting
7. [x] cloudfront_distribution_enabled
8. [x] datasync_location_object_storage_expose_secret
9. [x] ecr_repository_policy_prohibit_public_access
10. [x] ecs_task_definition_no_host_pid_mode
11. [x] ecs_task_definition_container_non_privileged
12. [x] ecs_task_definition_container_readonly_root_filesystem
13. [x] eks_cluster_node_group_ssh_access_from_internet
14. [x] rds_global_cluster_encryption_enabled

### CODE REVIEW

File: query/apigateway.sp [hunk 1]
```diff
@@ -273,6 +273,50 @@ query "apigatewayv2_route_set_authorization_type" {
...
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'http_method') != 'OPTIONS' and (arguments ->> 'authorization') = 'NONE' and (arguments ->> 'api_key_required') is null then 'alarm'
+        when (arguments ->> 'http_method') != 'OPTIONS' and (arguments ->> 'authorization') = 'NONE' and (arguments ->> 'api_key_required') = 'false' then 'alarm'
+        else 'ok'
+      end status,
+      name || case
+        when (arguments ->> 'http_method') != 'OPTIONS' and (arguments ->> 'authorization') = 'NONE' and (arguments ->> 'api_key_required') is null then ' API Gateway method is not restricted'
```
Comment (madhushreeray30) [inline query/apigateway.sp:290]:
```suggestion
        when (arguments ->> 'http_method') != 'OPTIONS' and (arguments ->> 'authorization') = 'NONE' and (arguments ->> 'api_key_required') is null then ' does not restrict open access'
```

File: query/apigateway.sp [hunk 2]
```diff
@@ -273,6 +273,50 @@ query "apigatewayv2_route_set_authorization_type" {
...
+        when (arguments ->> 'http_method') != 'OPTIONS' and (arguments ->> 'authorization') = 'NONE' and (arguments ->> 'api_key_required') = 'false' then ' API Gateway method is not restricted'
```
Comment (madhushreeray30) [inline query/apigateway.sp:291]:
```suggestion
        when (arguments ->> 'http_method') != 'OPTIONS' and (arguments ->> 'authorization') = 'NONE' and (arguments ->> 'api_key_required') = 'false' then ' does not restrict open access'
```

File: query/apigateway.sp [hunk 3]
```diff
@@ -273,6 +273,50 @@ query "apigatewayv2_route_set_authorization_type" {
...
+        else ' API Gateway method is restrictive with http_method as ' || (arguments ->> 'http_method') || ', authorization as ' || (arguments ->> 'authorization') || ' and api_key_required as ' || (arguments ->> 'api_key_required')
```
Comment (madhushreeray30) [inline query/apigateway.sp:292]:
```suggestion
        else ' is restrictive with http_method as ' || (arguments ->> 'http_method') || ', authorization as ' || (arguments ->> 'authorization') || ' and api_key_required as ' || (arguments ->> 'api_key_required')
```

File: query/apigateway.sp [hunk 4]
```diff
@@ -273,6 +273,50 @@ query "apigatewayv2_route_set_authorization_type" {
...
+query "apigateway_domain_name_use_latest_tls" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'security_policy') is null or (arguments ->> 'security_policy') = 'TLS_1_2' then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments ->> 'security_policy') is null or (arguments ->> 'security_policy') = 'TLS_1_2' then ' API Gateway Domain uses latest TLS security policy'
```
Comment (madhushreeray30) [inline query/apigateway.sp:312]:
```suggestion
        when (arguments ->> 'security_policy') is null or (arguments ->> 'security_policy') = 'TLS_1_2' then ' uses latest TLS security policy'
```

File: query/apigateway.sp [hunk 5]
```diff
@@ -273,6 +273,50 @@ query "apigatewayv2_route_set_authorization_type" {
...
+        else ' API Gateway Domain not using latest TLS security policy'
```
Comment (madhushreeray30) [inline query/apigateway.sp:313]:
```suggestion
        else ' does not use latest TLS security policy'
```

File: query/cloudfront.sp [hunk 6]
```diff
@@ -180,3 +209,52 @@ query "cloudfront_protocol_version_is_low" {
...
+        and (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'preload')::boolean
+        and (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'override')::boolean
+        then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'access_control_max_age_sec')::integer = 31536000
+        and (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'include_subdomains')::boolean
+        and (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'preload')::boolean
+        and (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'override')::boolean then ' CloudFront response header policy enforces Strict Transport Security settings'
```
Comment (madhushreeray30) [inline query/cloudfront.sp:229]:
```suggestion
        and (arguments -> 'security_headers_config' -> 'strict_transport_security' ->> 'override')::boolean then ' enforces Strict Transport Security settings'
```

File: query/cloudfront.sp [hunk 7]
```diff
@@ -180,3 +209,52 @@ query "cloudfront_protocol_version_is_low" {
...
+        else ' CloudFront response header policy not enforcing Strict Transport Security settings'
```
Comment (madhushreeray30) [inline query/cloudfront.sp:230]:
```suggestion
        else ' does not enforce Strict Transport Security settings'
```

File: query/cloudfront.sp [hunk 8]
```diff
@@ -180,3 +209,52 @@ query "cloudfront_protocol_version_is_low" {
...
+query "cloudfront_distribution_enabled" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'enabled')::boolean then 'ok'
+        else 'info'
+      end status,
+      name || case
+        when(arguments ->> 'enabled')::boolean then ' cloudfront distribution enabled'
```
Comment (madhushreeray30) [inline query/cloudfront.sp:250]:
```suggestion
        when(arguments ->> 'enabled')::boolean then ' is enabled'
```

File: query/cloudfront.sp [hunk 9]
```diff
@@ -180,3 +209,52 @@ query "cloudfront_protocol_version_is_low" {
...
+        else ' cloudfront distribution disabled'
```
Comment (madhushreeray30) [inline query/cloudfront.sp:251]:
```suggestion
        else ' is disabled'
```

File: query/datasync.sp [hunk 10]
```diff
@@ -0,0 +1,20 @@
+query "datasync_location_object_storage_expose_secret" {
+  sql = <<-EOQ
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'secret_key') is null then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'secret_key') is null then ' DataSync location object storage does not expose any secret key details'
```
Comment (madhushreeray30) [inline query/datasync.sp:10]:
```suggestion
        when (arguments -> 'secret_key') is null then ' does not expose any secret key details'
```

File: query/datasync.sp [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+        else ' DataSync location object storage exposes secret key detail'
```
Comment (madhushreeray30) [inline query/datasync.sp:11]:
```suggestion
        else ' exposes secret key detail'
```

File: query/ec2.sp [hunk 12]
```diff
@@ -429,4 +429,31 @@ query "ec2_ami_encryption_enabled" {
...
+        when (arguments -> 'organizational_arn') is not null then 'info'
+        when (arguments -> 'organizational_unit_arn') is not null then 'info'
+        else 'alarm'
+      end as status,
+      name || case
+        when (arguments -> 'account_id') is not null then ' AMI launch permission is restrictive to account(s)'
+        when (arguments -> 'group') is not null then ' AMI launch permission is open to IAM group'
+        when (arguments -> 'organizational_arn') is not null then ' AMI launch permission is open to organization'
+        when (arguments -> 'organizational_unit_arn') is not null then ' AMI launch permission is open to organization unit'
+        else ' AMI launch permission is wide open'
```
Comment (madhushreeray30) [inline query/ec2.sp:450]:
```suggestion
        when (arguments -> 'account_id') is not null then ' is restrictive to account(s)'
        when (arguments -> 'group') is not null then ' is open to IAM group'
        when (arguments -> 'organizational_arn') is not null then ' is open to the organization'
        when (arguments -> 'organizational_unit_arn') is not null then ' is open to organization unit'
        else ' is wide open'
```

File: query/eks.sp [hunk 13]
```diff
@@ -108,3 +115,28 @@ query "eks_cluster_control_plane_logging_enabled" {
...
+      case
+        when (arguments -> 'remote_access') is null then 'ok'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is not null then 'ok'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is null then 'alarm'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'remote_access') is null then ' node group does not have implicit SSH access from 0.0.0.0/0'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is not null then ' node group SSH access restricted to security group (s)'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is null then ' node group has implicit SSH access from 0.0.0.0/0'
```
Comment (madhushreeray30) [inline query/eks.sp:132]:
```suggestion
        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is not null then ' SSH access restricted to security group(s)'
        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is null then ' has implicit SSH access from 0.0.0.0/0'
```

File: query/eks.sp [hunk 14]
```diff
@@ -108,3 +115,28 @@ query "eks_cluster_control_plane_logging_enabled" {
...
+        else ' node group has implicit SSH access from 0.0.0.0/0'
```
Comment (madhushreeray30) [inline query/eks.sp:133]:
```suggestion
        else ' has implicit SSH access from 0.0.0.0/0'
```

File: query/rds.sp [hunk 15]
```diff
@@ -882,4 +882,29 @@ query "rds_mysql_db_cluster_audit_logging_enabled" {
...
+        when (arguments ->> 'storage_encrypted')::boolean then 'ok'
+        when (arguments ->> 'source_db_cluster_identifier') is null then 'info'
+        when (arguments ->> 'storage_encrypted')::boolean and (arguments ->> 'source_db_cluster_identifier') is not null then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments ->> 'storage_encrypted')::boolean then ' DB cluster enctyprion enabled'
+        when (arguments ->> 'source_db_cluster_identifier') is null then ' DB cluster identifier is not provided of the primary global cluster creation'
+        when (arguments ->> 'storage_encrypted')::boolean and (arguments ->> 'source_db_cluster_identifier') is not null then ' DB cluster enctyprion enabled'
+        else ' DB cluster enctyprion disabled'
```
Comment (madhushreeray30) [inline query/rds.sp:901]:
```suggestion
        when (arguments ->> 'storage_encrypted')::boolean and (arguments ->> 'source_db_cluster_identifier') is not null then ' encryption enabled'
        else ' encryption disabled'
```

File: query/rds.sp [hunk 16]
```diff
@@ -882,4 +882,29 @@ query "rds_mysql_db_cluster_audit_logging_enabled" {
...
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'storage_encrypted')::boolean then 'ok'
+        when (arguments ->> 'source_db_cluster_identifier') is null then 'info'
+        when (arguments ->> 'storage_encrypted')::boolean and (arguments ->> 'source_db_cluster_identifier') is not null then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments ->> 'storage_encrypted')::boolean then ' DB cluster enctyprion enabled'
```
Comment (madhushreeray30) [inline query/rds.sp:898]:
```suggestion
        when (arguments ->> 'storage_encrypted')::boolean then ' DB cluster encryption enabled'
```

**Review by madhushreeray30 — COMMENTED** [review]
Please take a look at the comments, thanks!

File: controls/rds.sp [hunk 17]
```diff
@@ -416,3 +417,11 @@ control "rds_mysql_db_cluster_audit_logging_enabled" {
 
   tags = local.rds_compliance_common_tags
 }
+
+control "rds_global_cluster_encryption_enabled" {
```
Comment (misraved) [inline controls/rds.sp:421]:
```suggestion
control "rds_db_global_cluster_encryption_enabled" {
```

File: controls/rds.sp [hunk 18]
```diff
@@ -416,3 +417,11 @@ control "rds_mysql_db_cluster_audit_logging_enabled" {
...
+  title       = "RDS DB global cluster (MySQl & PostgreSQL) should have encryption enabled"
+  description = "This control checks whether RDS DB cluster (MySQl & PostgreSQL) encryption is enabled."
+  query       = query.rds_global_cluster_encryption_enabled
```
Comment (misraved) [inline controls/rds.sp:424]:
```suggestion
  query       = query.rds_db_global_cluster_encryption_enabled
```

File: query/cloudfront.sp [hunk 19]
```diff
@@ -109,32 +109,61 @@ query "cloudfront_distribution_origin_access_identity_enabled" {
...
+        when (t.name is null ) and ((arguments -> 'origin' ->> 'domain_name') not like '%aws_s3_bucket%') then 'skip'
         else 'ok'
       end as status,
-      case
+      a.name || case
+        when (arguments -> 'origin') is null then ' origins not defined'
+        when (arguments -> 'origin' ->> 'domain_name' ) like '%aws_s3_bucket%' and (( not((arguments -> 'origin' -> 's3_origin_config' ->> 'origin_access_identity') = '')) and (arguments -> 'origin' -> 's3_origin_config' -> 'origin_access_identity') is not null) then ' origin access identity configured'
+        when (arguments -> 'origin' ->> 'domain_name' ) like '%aws_s3_bucket%' and (((arguments -> 'origin' -> 's3_origin_config' ->> 'origin_access_identity') = '') or ((arguments -> 'origin' -> 's3_origin_config') is null)) then ' origin access identity not configured'
         when b.name is not null then ' origin access identity not configured'
+        when (t.name is null ) and ((arguments -> 'origin' ->> 'domain_name') not like '%aws_s3_bucket%') then ' origin type is not s3'
```
Comment (misraved) [inline query/cloudfront.sp:159]:
```suggestion
        when (t.name is null ) and ((arguments -> 'origin' ->> 'domain_name') not like '%aws_s3_bucket%') then ' origin type is not S3'
```

File: query/eks.sp [hunk 20]
```diff
@@ -108,3 +115,28 @@ query "eks_cluster_control_plane_logging_enabled" {
...
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'remote_access') is null then 'ok'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is not null then 'ok'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is null then 'alarm'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'remote_access') is null then ' node group does not have implicit SSH access from 0.0.0.0/0'
+        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is not null then ' SSH access restricted to security group (s)'
```
Comment (misraved) [inline query/eks.sp:131]:
```suggestion
        when (arguments -> 'remote_access' ->> 'ec2_ssh_key') is not null and (arguments -> 'remote_access' ->> 'source_security_group_ids') is not null then ' SSH access restricted to security group(s)'
```

**Review by misraved — COMMENTED** [review]
@madhushreeray30 please take a look at the minor review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**